### PR TITLE
chore(sdk): Use pipeline_spec proto from kfp-pipeline-spec package

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**
(As said in the title)

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: sdk/python/kfp/__init__.py [hunk 1]
```diff
@@ -12,11 +12,15 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-__version__ = '1.1.0'
+__version__ = '1.1.1rc1'
```
Comment (chensun) [inline sdk/python/kfp/__init__.py:15]: This is temporary, won't actually release this version.

File: sdk/python/kfp/__init__.py [hunk 2]
```diff
@@ -12,7 +12,11 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-__version__ = '1.1.0'
+# `kfp` is a namespace package.
```
Comment (Ark-kun) [inline sdk/python/kfp/__init__.py:15]: Are you sure kfp is a namespace package?

I'm also not sure the following line is needed for a namespace package.

**Comment by Ark-kun** [comment]
What is the main motivation to move the spec into its own package? It seems to be closely tied to the v2 compiler.

**Pull request merged**